sqlmesh/core/table_diff.py

```diff
@@ -106,7 +107,6 @@ def __init__(
         target: TableName,
         on: t.List[str] | exp.Condition,
         where: t.Optional[str | exp.Condition] = None,
-        dialect: DialectType = None,
```

Comment:
This was unused - not sure if we meant this to be part of the API, given that we have the engine adapter in here. Happy to revert this change if I missed a use case, cc @tobymao please take a look.